schema/rdl_schema_0.1.json

```
@@ -1710,13 +1552,502 @@
+          "title": "Socio-economic index category",
+          "type": "object",
+          "description": "The category used to classify the socio-economic category vulnerable to the hazard.",
+          "$ref": "#/$defs/Se_category"
+        }
+      },
+      "minProperties": 1
+    },
+    "Vulnerability_function": {
+      "title": "Vulnerability function",
+      "type": "object",
+      "description": "Details of a vulnerability function used to calculate the impact of the hazard.",
+      "properties": {
+        "approach": {
+          "title": "Impact function approach",
+          "type": "string",
+          "description": "The approach the impact function is based upon, taken from the closed [function approach codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#function_approach).",
+          "enum": [
+            "analytical",
+            "empirical",
+            "hybrid",
+            "judgement"
+          ],
+          "codelist": "function_approach.csv",
+          "openCodelist": false
+        },
+        "relationship": {
+          "title": "Impact relationship type",
+          "type": "string",
+          "description": "The type of function relationships used to calculate the impact values, taken from the closed [relationship type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#relationship_type).",
+          "enum": [
+            "discrete values",
+            "mathematical function (bespoke)",
+            "mathematical function (parametric)"
+          ],
+          "codelist": "relationship_type.csv",
+          "openCodelist": false
+        }
+      },
+      "minProperties": 1
+    },
+    "Fragility_function": {
+      "title": "Fragility function",
+      "type": "object",
+      "description": "Details of a fragility function used to calculate the impact of the hazard.",
+      "properties": {
+        "approach": {
+          "title": "Impact function approach",
+          "type": "string",
+          "description": "The approach the impact function is based upon, taken from the closed [function approach codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#function_approach).",
+          "enum": [
+            "analytical",
+            "empirical",
+            "hybrid",
+            "judgement"
+          ],
+          "codelist": "function_approach.csv",
+          "openCodelist": false
+        },
+        "relationship": {
+          "title": "Impact relationship type",
+          "type": "string",
+          "description": "The type of function relationships used to calculate the impact values, taken from the closed [relationship type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#relationship_type).",
+          "enum": [
+            "discrete values",
+            "mathematical function (bespoke)",
+            "mathematical function (parametric)"
```

Update per values in CSV file:

```suggestion
            "discrete",
            "math_bespoke",
            "math_parametric"
```